Artesian/_GMEPublicOffers/GMEPOfferQuery.py

```diff
@@ -13,57 +17,125 @@
 import asyncio
 import itertools
 import urllib
+from __future__ import annotations
 
 class _GMEPOfferQuery:
     __routePrefix = "extract"
-    def __init__(self, client, requestExecutor, partitionStrategy):
+    def __init__(self, client: _ClientsExecutor , requestExecutor: RequestExecutor, partitionStrategy: DefaultPartitionStrategy) -> None: 
+        """ Inits _GME Public Offer Query
+        
+        Args:
+                client
+                
+                requestExecutor
+                
+                partitionStrategy. 
+
+        """
+
         queryParameters = GMEPOfferQueryParameters(None,ExtractionRangeConfig(), None, None, None, None, None, None, None, None, None, None, None )
         self._queryParameters = queryParameters
         self.__client = client
         self.__executor = requestExecutor
         self.__partition= partitionStrategy
 
-    def forScope(self, scope):
-        self._queryParameters.scope = scope
+    def forScope(self, scope: Scope) -> _GMEPOfferQuery:
+        """ Set the scopes to be queried.
+        
+            Returns:
+                GMEPublicOfferQuery."""
+        self._queryParameters.scope = scope 
         return self
-    def forStatus(self, status):
+    def forStatus(self, status: Status) -> _GMEPOfferQuery:
+        """ Set the status to be queried.
+        
+            Returns:
+                GMEPublicOfferQuery."""
         self._queryParameters.status = status
         return self
-    def forUnitType(self, unitType):
+    def forUnitType(self, unitType: UnitType) -> _GMEPOfferQuery:
+        """ Set the unit types to be queried.
+        
+            Returns:
+                GMEPublicOfferQuery."""
         self._queryParameters.unitType = unitType
         return self
-    def forDate(self, date):
+    def forDate(self, date: str) -> _GMEPOfferQuery:
+        """ Set the date to be queried.
+        
+            Returns:
+                GMEPublicOfferQuery."""
         self._queryParameters.extractionRangeSelectionConfig.date = date
         return self
-    def forUnit(self, unit):
+    def forUnit(self, unit: int) -> _GMEPOfferQuery:
```

Comment:
missing Args documentation.
I'm quite sure unit is not an 'int'. Please check with @MicheleNan